### DISCUSSION TITLE
PremadeOverlays

### DISCUSSION THREAD

**Discussion by IzzySoft** [Mahmud0808/Iconify#86] [General]
One of my scanners just reported "embedded APKs" (a new check recently added) with your latest release – and indeed there are quite some (68) APKs inside the APK. I've checked the repo, and there they are as APKs inside assets. There seems no danger with these here in this case, but it still raises some questions:

* is the source of those APKs also F/LOSS – and if so, where can it be found?
* as APKs inside APKs are not, well, what one usually does, such a finding always raises suspicion first. And while I guess you'll have your reasons and already thought it through: is there a "less suspicious" alternative to embedding APKs?

No offense meant at all – just seeking clarification: if an app comes with "root powers", some extra caution cannot hurt, right? So thanks in advance for any "enlightenment" you can give :smiley:

**Comment by Mahmud0808** [comment]
These are Runtime Resource Overlays. They have nothing inside but just a bunch of resources. The reason to use Premade APKs is to reduce the time of building overlays. There is ofcourse alternative way to not use premade APKs, but this is the only solution I found that's the fastest and works on every rom without any hassle. The source can be made public ofcourse, it's currently in my private repo which is very messy rn.

**Comment by IzzySoft** [comment]
Thanks for the swift answer, @Mahmud0808 – and yes, that's basically what I guessed already. May I suggest to add a note on that to the description or at least the Readme, for transparency? And yes, please see if you can make the source of the embedded APKs available openly as well. Feel encouraged to take this as motivation for "cleaning some mess" :see_no_evil:

**Comment by Mahmud0808** [comment]
Source can be found [Here](https://github.com/Mahmud0808/IconifyComponent)

Note is added in assets/PremadeOverlays/cheatsheet file.

**Comment by IzzySoft** [comment]
Thanks a lot! Mind adding a pointer to the app description as well? Guess it could go as a 4th point into the disclaimer, like

```html
<li>Note: PremadeOverlays are used as embedded APKs to reduce the overlay compilation time. For details, please <a href='https://github.com/Mahmud0808/Iconify/blob/beta/app/src/main/assets/PremadeOverlays/cheatsheet'>see here</a>.</li>
```

**Reply by Mahmud0808** [reply]
Done :)

**Reply by IzzySoft** [reply]
Thanks a lot! LGTM then :smiley:
